core/src/templates/components/card/card.twig

```diff
@@ -3,78 +3,94 @@
  * @file
  * Card Component.
  *
- * Basic building block component with two variants
- *   - Horizontal: An image beside text content
+ * Basic building block component with the following variants:
  *   - Vertical (default): An image above text content
+ *   - Horizontal: An image beside text content
+ *   - Link: Whole card is a link with no button or call to action link
  *
  * Available variables:
  * - attributes: For additional HTML attributes not already provided.
- * - modifier_class: Additional css classes to change look and behaviour.
- * -- su-card--horizontal
+ * - modifier_class: Additional CSS classes to change look and behaviour.
  * - card_link: The href value for the card. A single card is only allowed to link to one URL.
  * - card_image: A full HTML entity for an image.
- * - card_super_headline: Text heading
+ * - card_super_headline: Text super headline. Often used for metadata such as category.
  * - card_headline: Text heading
  * - card_body: An open variable for anything. Generally this would be for
  *     WYSIWYG editor contents.
- * - card_cta_attributes: Any html attributes on the cta link that need to be
+ * - card_cta_attributes: Any html attributes on the link that need to be
  *     added. For example: _target.
  * - card_cta_label: The clickable text for the CTA link
  * - card_button_label: The text label of the button component.
  */
 #}
-<div{{ attributes }} class="su-card {{ modifier_class }}">
 
-  {# Card Image #}
-  {% if card_image is not empty -%}
-    {{ card_image }}
+{# For variant where whole card is a link, change the outer wrapper from <div> to <a> #}
+{% if "su-card--link" in modifier_class %}
+  {% set allow_links = false %}
+{% else %}
+  {% set allow_links = true %}
+{%- endif -%}
+
+<div {{ attributes }} class="su-card {{ modifier_class }}">
+  {%- if allow_links == false %}
+    <a href="{{ card_link }}" {{ card_cta_attributes }}>
```

Comment:
I like this approach as re-uses the variable but it has a couple of trade offs.

You can only use the outer most linking of the card if the card--link variant is passed through. I would suspect we could have more than one use case or variant that will allow for full linking and it would require us to add all of the variants to this template to make them work. What if, instead of depending on the class name to trigger the field, it was its own variable? eg: `card_link_wrapper` or something. Or perhaps, instead of the variant name check, add another boolean variable that is the toggle. eg: Add a variable that is `outer_link` and if set to true, render around, if set to false, render the button.

@kerri-augenstein Would there, should there, could there, be a use case where the whole card was wrapped and the CTA button still there? You would only be able to navigate to the outer-most-wrapper link but visually, we could still render a button without a link.

Reply:
@sherakama , I'm fine either way. This might be more of a @JBCSU question 🤔

Reply:
> I like this approach as re-uses the variable but it has a couple of trade offs.
> 
> You can only use the outer most linking of the card if the card--link variant is passed through. I would suspect we could have more than one use case or variant that will allow for full linking and it would require us to add all of the variants to this template to make them work. What if, instead of depending on the class name to trigger the field, it was its own variable? eg: `card_link_wrapper` or something. Or perhaps, instead of the variant name check, add another boolean variable that is the toggle. eg: Add a variable that is `outer_link` and if set to true, render around, if set to false, render the button.
> 
> @kerri-augenstein Would there, should there, could there, be a use case where the whole card was wrapped and the CTA button still there? You would only be able to navigate to the outer-most-wrapper link but visually, we could still render a button without a link.

Actually I'll just do that - extra option to not use the class name. That will make it more flexible.

Reply:
Hm....this is more complicated than I thought, because there's some styling on su-card that needs to be removed if there is an outer_link, also for the horizontal variant, if there is an outer_link, the grid styling on the parent container needs to be removed and put on outer_link instead. Could we leave that till later?

Reply:
We might come back to this later and add a boolean variable.